main.go

```diff
@@ -138,20 +146,74 @@ func readJSON(file string) (string, error) {
 	return "", nil
 }
 
-// Default 3LO authorization handler. Prints the authorization URL on stdout
+// 3LO authorization handler. Determines what algorithm to use
+// to get the authorization code.
+//
+// Note that the "state" parameter is used to prevent CSRF attacks.
+func cmdAuthorizationHandler(state string, consentSettings util.ConsentPageSettings,
+	authCodeServer *util.AuthorizationCodeServer) authhandler.AuthorizationHandler {
+	return func(authCodeURL string) (string, string, error) {
+
+		decodedValue, _ := url.ParseQuery(authCodeURL)
+		redirectURL := decodedValue.Get("redirect_uri")
+
+		if strings.Contains(redirectURL, "localhost") {
+			return cmdAuthorizationLoopback(authCodeURL, consentSettings, authCodeServer)
+		}
+
+		return cmdAuthorizationInteractive(state, authCodeURL)
+	}
+}
+
+// Interactive 3LO authorization. Prints the authorization URL on stdout
 // and reads the authorization code from stdin.
 //
 // Note that the "state" parameter is used to prevent CSRF attacks.
-// For convenience, CmdAuthorizationHandler returns a pre-configured state
+// For convenience, cmdAuthorizationInteractive returns a pre-configured state
 // instead of requiring the user to copy it from the browser.
-func cmdAuthorizationHandler(state string) authhandler.AuthorizationHandler {
-	return func(authCodeURL string) (string, string, error) {
-		fmt.Printf("Go to the following link in your browser:\n\n   %s\n\n", authCodeURL)
-		fmt.Println("Enter authorization code:")
-		var code string
-		fmt.Scanln(&code)
-		return code, state, nil
+func cmdAuthorizationInteractive(state string, authCodeURL string) (string, string, error) {
+	fmt.Printf("Go to the following link in your browser:\n\n   %s\n\n", authCodeURL)
+	fmt.Println("Enter authorization code:")
+	var code string
+	fmt.Scanln(&code)
+	return code, state, nil
+}
+
+// Loopback 3LO authorization. Prints the authorization URL on stdout
+// and opens a new browser window and redirects it to the authCodeURL.
+// It then reads the authorization code from a localhost endpoint.
+//
+// Note that the "state" parameter is used to prevent CSRF attacks.
+// For convenience, cmdAuthorizationLoopback returns the state
+// associated with the authorization code sent to the localhost.
```

Comment:
In line 187-188 I am trying to communicate that the 'state' is the value associated with the authorization code that is sent over to the localhost. Maybe this is not too clear?